### DISCUSSION TITLE
Building user documentation locally

### DISCUSSION THREAD

**Discussion by capsulecorplab** [nasa/fprime#2364] [Q&A]
Hello fprime-community,
I'd like to build the fprime user documentation, as to be able to add it to the [fprime-workspace-image](https://github.com/capsulecorplab/fprime-workspace-image) virtual environment I've been working on. If someone could point me to what doc tools I need and how to locally run the user docs build, that'd be much appreciated :)

**Comment by thomas-bc** [comment]
@capsulecorplab nice!

This is the workflow that build our docs: https://github.com/nasa/fprime/blob/devel/.github/workflows/autodocs.yml

So the main script you'll want to look at is https://github.com/nasa/fprime/blob/devel/docs/doxygen/generate_docs.bash

Let me know if you have any other questions!

**Reply by capsulecorplab** [reply]
Thanks @thomas-bc ! I tried running the `generate_docs.bash` script but was only able to generate the API documentation for some reason

**Reply by LeStarch** [reply]
That is all that it does.  We use Github pages (and thus Jekyll) to convert the MD files under docs to the final HTML.

**Reply by capsulecorplab** [reply]
@LeStarch as in the md files are converted to HTML with Jekyll and manually uploaded to GitHub pages? What Jekyll command should I run to produce the same output locally?

**Reply by thomas-bc** [reply]
@capsulecorplab I have not tried it myself, but this may be what you're looking for: https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/testing-your-github-pages-site-locally-with-jekyll

**Reply by LeStarch** [reply]
@capsulecorplab, no, Github Pages does this automatically.  The above link may work, but I've not tried it myself either.

**Reply by capsulecorplab** [reply]
So, following the steps from https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/testing-your-github-pages-site-locally-with-jekyll#building-your-site-locally, I installed bundler (on a branch) in my environment and attempted to run the bundler command to build/serve with jekyll, but it appears it's expecting a Gemfile or .bundle/ directory. Should I have skipped installing bundler and installed jekyll directly or am I in the wrong directory?

![Screenshot from 2023-11-15 01-17-06](https://github.com/nasa/fprime/assets/14095576/f2bd9758-054c-4dfe-a872-79449abc278f)

**Reply by LeStarch** [reply]
About all the information I can give you is that we use the "docs" folder as our source.  (We technically build off an auto-documentation branch too, but I doubt that would make a difference).

**Reply by capsulecorplab** [reply]
So, I managed to install jekyll to run a build of the docs, but it looks like it's not quite rendering properly. Any idea as to what I'm doing wrong? ![Screenshot from 2023-11-21 15-53-09](https://github.com/nasa/fprime/assets/14095576/33e5f92d-5d6b-47fe-88a0-244a5b13d9b5)
Here's the line in my ansible playbook where I'm running the jekyll build https://github.com/capsulecorplab/fprime-workspace-image/blob/52aff00fee18933e301d1bcebe39cc5de3399415/playbook.yaml#L332

**Reply by thomas-bc** [reply]
Is it picking up the correct theme? Not sure what all the `Your awesome title` things are doing there, and the lack of formatting/images/colors etc... kind of makes me wonder if it may have picked up the wrong theme, or no theme at all. But again, we let GitHub Pages do everything for us so we're not the best resource to answer these questions. Maybe the jekyll folks?

The theme is configured in this file: https://github.com/nasa/fprime/blob/devel/docs/_config.yml

**Reply by capsulecorplab** [reply]
I haven't had time learn Jekyll themes yet, but created a (draft) PR in case anyone in the community would like a preview https://github.com/capsulecorplab/fprime-workspace-image/pull/35

**Reply by capsulecorplab** [reply]
I've made another pass at attempting to install the fprime jekyll theme with no luck, but I also noticed the link to the tutorials don't work. Closing my [PR](https://github.com/capsulecorplab/fprime-workspace-image/pull/35) for now, as it appears the fprime/docs codebase would need to undergo refactoring in order to support a local build of the documentation.
